src/modules/billing/jobs/create-charge.js

```diff
@@ -82,20 +88,25 @@ const handler = async job => {
     // Note: the await is needed to ensure any error is handled here
     return await updateBatchState(batch);
   } catch (err) {
-    // Always log and mark transaction as errored in DB
-    transactionsService.setErrorStatus(transactionId);
-    batchJob.logHandlingError(job, err);
-
-    // If not a client error, error the batch
-    if (!isClientError(err)) {
-      await batchService.setErrorStatus(batchId, BATCH_ERROR_CODE.failedToCreateCharge);
-      throw err;
+    const statusCode = get(err, 'statusCode');
+
+    // if the charge was created in the CM
+    if (statusCode === 409) {
+      batchJob.logHandlingError(job, err);
+      await transactionsService.updateWithChargeModuleResponse(transactionId, err.response.body);
+      return await updateBatchState(batch);
+    }
+    // if error code >= 400 and < 500 set transacti on status to error and continue
+    if (isClientError(err)) {
+      batchJob.logHandlingError(job, err);
+      transactionsService.setErrorStatus(transactionId);
+      return {
+        batch: job.data.batch
+      };
     }
+    // throw error to retry
+    throw new StateError(`Charge not created in CM for transaction ${transactionId} in batch ${batchId}`);
```

Comment:
I think we should possibly rethrow `err` here instead of creating a new error.  Otherwise e.g. if it's a DB error our side this will be masked.